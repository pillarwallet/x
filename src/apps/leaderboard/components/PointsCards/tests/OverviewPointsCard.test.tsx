import { render, screen } from '@testing-library/react';

// components
import {
  LeaderboardRankChange,
  LeaderboardTableData,
} from '../../../../../types/api';
import OverviewPointsCard from '../OverviewPointsCard';

const myAllTimeMergedMock: LeaderboardTableData = {
  totalPoints: 1234,
  totalAmountUsd: 10000,
  addresses: ['0xabc123456789def'],
  totalGas: 98765,
};

const myWeeklyMergedMock: LeaderboardTableData = {
  totalPoints: 345,
  totalAmountUsd: 200,
  addresses: ['0xabc123456789def'],
  completedSwap: true,
  totalGas: 78765,
  rankChange: LeaderboardRankChange.INCREASED,
};

describe('OverviewPointsCard', () => {
  it('matches snapshot with full data', () => {
    const { container } = render(
      <OverviewPointsCard
        myAllTimeMerged={{ entry: myAllTimeMergedMock, index: 4 }}
        myWeeklyMerged={{ entry: myWeeklyMergedMock, index: 9 }}
        timeTab="all"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with empty data', () => {
    const { container } = render(
      <OverviewPointsCard
        myAllTimeMerged={{ entry: undefined, index: -1 }}
        myWeeklyMerged={{ entry: undefined, index: -1 }}
        timeTab="all"
      />
    );
    expect(container).toMatchSnapshot();
  });

  it('renders all sections with valid data', () => {
    render(
      <OverviewPointsCard
        myAllTimeMerged={{ entry: myAllTimeMergedMock, index: 0 }}
        myWeeklyMerged={{ entry: myWeeklyMergedMock, index: 2 }}
        timeTab="all"
      />
    );

    expect(screen.getByText('My PX Overview')).toBeInTheDocument();
    expect(screen.getByText('PX Points')).toBeInTheDocument();
    expect(screen.getByText('1,234')).toBeInTheDocument();
    expect(screen.getByText('Current Rank')).toBeInTheDocument();
    expect(screen.getByText('1')).toBeInTheDocument();
    expect(screen.getByText('Weekly')).toBeInTheDocument();
    expect(screen.getByText('Weekly Rank')).toBeInTheDocument();
    expect(screen.getByText('3')).toBeInTheDocument();
    expect(screen.getByText('345')).toBeInTheDocument();
    expect(screen.getByText('Earned This Week')).toBeInTheDocument();
  });

  it('shows fallback for missing data', () => {
    render(
      <OverviewPointsCard
        myAllTimeMerged={{ entry: undefined, index: -1 }}
        myWeeklyMerged={{ entry: undefined, index: -1 }}
        timeTab="all"
      />
    );

    expect(screen.getAllByText('-').length).toBe(4);
  });
});
