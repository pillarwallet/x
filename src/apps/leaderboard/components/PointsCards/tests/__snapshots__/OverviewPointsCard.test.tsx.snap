// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OverviewPointsCard > matches snapshot with empty data 1`] = `
<div>
  <div
    class="flex flex-col w-full h-full rounded-[10px] bg-purple_medium/[.05] p-3 gap-3"
  >
    <p
      class="text-sm font-medium font-normal"
    >
      My PX Overview
    </p>
    <div
      class="flex flex-col w-full gap-2.5"
    >
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="pillarx-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='6'%20viewBox='0%200%2016%206'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_1577_2383)'%3e%3cpath%20opacity='0.5'%20d='M15.5213%201.57674H16.0001V0.600006H15.5213C15.081%200.600006%2014.6584%200.770336%2014.3599%201.06575L13.6152%201.80829C13.5236%201.89877%2013.3935%201.952%2013.2605%201.952C13.1275%201.952%2012.9975%201.89877%2012.9059%201.80829L12.1612%201.06575C11.8627%200.770336%2011.4401%200.600006%2010.9997%200.600006H10.5269V1.57674H10.9997C11.1357%201.57674%2011.2628%201.62997%2011.3544%201.72046L12.0784%202.4417L11.3544%203.16294C11.2628%203.25343%2011.1327%203.30666%2010.9968%203.30666H10.5239V4.2834H10.9968C11.4371%204.2834%2011.8597%204.11307%2012.1582%203.81765L12.9029%203.07512C12.9946%202.98463%2013.1246%202.9314%2013.2576%202.9314C13.3906%202.9314%2013.5206%202.98463%2013.6122%203.07512L14.3599%203.81765C14.6584%204.11307%2015.081%204.2834%2015.5213%204.2834H16.0001V3.30666H15.5213C15.3854%203.30666%2015.2583%203.25343%2015.1667%203.16294L14.4427%202.4417L15.1667%201.72046C15.2583%201.62997%2015.3884%201.57674%2015.5213%201.57674Z'%20fill='%238A77FF'/%3e%3cpath%20d='M1.03333%205.81735H0.0610335C0.0285252%205.81735%200.00488281%205.7934%200.00488281%205.76679V4.32962C0.00488281%203.29167%200.941713%202.44801%202.09428%202.44801C2.43119%202.44801%202.76513%202.52253%203.06362%202.66092L6.43266%204.30301C6.82276%204.49463%207.28379%204.49197%207.66798%204.29769C8.08468%204.08744%208.32997%203.67758%208.32997%203.24643V2.5518C8.32997%201.91306%207.75368%201.39409%207.04441%201.39409H0.0639888C0.0314805%201.39409%200.00783811%201.37014%200.00783811%201.34352V0.47058C0.00783811%200.441305%200.0344358%200.420013%200.0639888%200.420013H7.04145C8.35065%200.420013%209.41161%201.37546%209.41161%202.55446V3.23578C9.41161%204.01292%208.97422%204.74747%208.22948%205.13603C7.48475%205.5246%206.64544%205.52194%205.91253%205.16531L2.55235%203.52588C2.41641%203.462%202.25682%203.42741%202.09428%203.42741C1.54164%203.42741%201.08948%203.83194%201.08948%204.33229V5.76945C1.08948%205.79872%201.06288%205.82001%201.03333%205.82001V5.81735Z'%20fill='%238A77FF'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_1577_2383'%3e%3crect%20width='16'%20height='6'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            PX Points
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          -
           
          <span
            class="font-semibold text-white/[.5]"
          >
            PX
          </span>
        </p>
      </div>
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="current-rank-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_1577_2405)'%3e%3cpath%20opacity='0.4'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M2.3999%2014.4C2.3999%2013.9582%202.71331%2013.6%203.0999%2013.6H12.8999C13.2865%2013.6%2013.5999%2013.9582%2013.5999%2014.4C13.5999%2014.8418%2013.2865%2015.2%2012.8999%2015.2H3.0999C2.71331%2015.2%202.3999%2014.8418%202.3999%2014.4Z'%20fill='%238A77FF'/%3e%3cpath%20d='M6.96961%201.51463C7.14292%201.0818%207.54939%200.799988%207.99999%200.799988C8.45074%200.799988%208.85699%201.08193%209.03038%201.51455L10.5707%205.36044C10.6669%205.60075%2010.5977%205.87835%2010.4012%206.03961C10.1842%206.21776%2010.0465%206.49196%2010.0465%206.79999C10.0465%207.33449%2010.4629%207.76773%2010.9767%207.76773C11.4904%207.76773%2011.9069%207.33449%2011.9069%206.79999C11.9069%206.69688%2011.8916%206.59859%2011.8636%206.50683C11.7957%206.28393%2011.8629%206.04036%2012.0341%205.88888L13.9344%204.20787C14.4266%203.7724%2015.1643%203.79633%2015.6284%204.26555C15.9609%204.6017%2016.0849%205.10344%2015.9405%205.56561L15.939%205.57058L14.0787%2011.3473C13.7992%2012.2151%2013.0179%2012.8%2012.1382%2012.8H3.86179C2.98205%2012.8%202.20075%2012.2151%201.92129%2011.3473L0.0594687%205.5656C-0.0849019%205.10344%200.0391005%204.6017%200.371607%204.26555C0.835751%203.79633%201.57328%203.77241%202.06554%204.20786L3.96588%205.88885C4.13714%206.04033%204.20432%206.28392%204.13633%206.50682C4.10835%206.59856%204.09304%206.69685%204.09304%206.79999C4.09304%207.33449%204.50951%207.76773%205.02327%207.76773C5.53702%207.76773%205.95349%207.33449%205.95349%206.79999C5.95349%206.49198%205.81581%206.21777%205.59875%206.03959C5.4023%205.87833%205.33304%205.60075%205.42928%205.36043L6.96961%201.51463Z'%20fill='%238A77FF'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_1577_2405'%3e%3crect%20width='16'%20height='16'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Current Rank
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          <span
            class="font-semibold text-white/[.5]"
          >
            #
          </span>
          -
        </p>
      </div>
    </div>
    <p
      class="text-sm font-medium font-normal"
    >
      Weekly
    </p>
    <div
      class="flex flex-col w-full gap-2.5"
    >
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="pillarx-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M7.11805%201.96958C7.55465%201.12122%208.77852%201.12122%209.21512%201.96958L10.6861%204.8282C10.7105%204.87563%2010.7564%204.90864%2010.8095%204.91702L14.007%205.42134C14.9554%205.57093%2015.3333%206.72184%2014.6551%207.39531L12.3661%209.66831C12.3281%209.70598%2012.3107%209.75924%2012.3191%209.81178L12.8236%2012.9841C12.9731%2013.9243%2011.9832%2014.6359%2011.1269%2014.2039L8.24292%2012.7489C8.19498%2012.7247%208.13819%2012.7247%208.09025%2012.7489L5.20627%2014.2039C4.34995%2014.6359%203.36008%2013.9243%203.5096%2012.9841L4.01414%209.81178C4.0225%209.75924%204.00502%209.70598%203.96711%209.66831L1.67804%207.39531C0.999871%206.72184%201.37779%205.57093%202.32619%205.42134L5.5237%204.91702C5.57678%204.90864%205.62263%204.87563%205.64704%204.8282L7.11805%201.96958Z'%20fill='%238A77FF'/%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Weekly Rank
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          <span
            class="font-semibold text-white/[.5]"
          >
            #
          </span>
          -
        </p>
      </div>
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="current-rank-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M7%201.16669C8.5474%201.16669%209.80427%202.40717%209.832%203.94794H4.16797C4.19576%202.40717%205.4526%201.16669%207%201.16669Z'%20fill='%238A77FF'/%3e%3cpath%20opacity='0.4'%20d='M3.63086%203.05467C3.75694%202.60425%203.97053%202.19071%204.25261%201.83331H3.44141C2.3694%201.83331%201.50011%202.65423%201.5%203.66665V10.3763C1.49998%2011.2849%201.49999%2012.0178%201.57747%2012.5944C1.65815%2013.1944%201.83121%2013.6996%202.23242%2014.1009C2.62076%2014.4508%203.17638%2014.6802%203.73893%2014.7558C4.31554%2014.8334%205.04833%2014.8333%205.95703%2014.8333H10.043C10.9517%2014.8333%2011.6845%2014.8334%2012.2611%2014.7558C12.8611%2014.6752%2013.4392%2014.426%2013.8405%2014.0247C14.1903%2013.6364%2014.3469%2013.1569%2014.4225%2012.5944C14.4651%2012.2778%2014.4829%2011.914%2014.4915%2011.5H12.6667C12.4236%2011.5%2012.133%2011.4848%2011.9453%2011.4486C11.46%2011.3184%2011.0739%2010.9552%2010.9134%2010.4837C10.8581%2010.2769%2010.8333%209.94431%2010.8333%209.66665C10.8333%209.38891%2010.8581%209.05631%2010.9134%208.84958C11.074%208.37811%2011.46%208.01478%2011.9453%207.88471C12.133%207.84845%2012.4236%207.83331%2012.6667%207.83331H14.4915C14.4829%207.41931%2014.4651%207.05551%2014.4225%206.73891C14.3469%206.17642%2014.1903%205.69688%2013.8405%205.30857C13.4935%204.96163%2012.9959%204.70937%2012.4993%204.61587C12.4983%204.28285%2012.4917%204.01122%2012.4427%203.77147L12.4115%203.64516C12.1871%202.85445%2011.5601%202.22588%2010.7467%201.96417L10.5814%201.91665C10.3413%201.85592%2010.0743%201.84037%209.74933%201.83592C10.0536%202.22214%2010.2774%202.67389%2010.3978%203.16665C10.8171%203.34961%2011.1235%203.73991%2011.1869%204.21027C11.1986%204.29749%2011.1986%204.40137%2011.1986%204.60619V4.61521H3.57422C3.14344%204.61504%202.79427%204.26544%202.79427%203.83461C2.79451%203.40399%203.14358%203.05485%203.57422%203.05467H3.63086Z'%20fill='%238A77FF'/%3e%3cpath%20d='M14.4999%2010.5H12.6666C12.4098%2010.5%2012.248%2010.4908%2012.248%2010.4908C12.1049%2010.4999%2011.8502%2010.2998%2011.8502%2010.1295C11.8501%2010.0803%2011.8333%2010.0085%2011.8333%209.66665C11.8333%209.32458%2011.8375%209.25131%2011.8502%209.20378C11.8503%209.06751%2012.0859%208.84251%2012.248%208.84245C12.3046%208.83585%2012.4098%208.83331%2012.6666%208.83331H14.4999V10.5Z'%20fill='%238A77FF'/%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Earned This Week
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          -
           
          <span
            class="font-semibold text-white/[.5]"
          >
            PX
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`OverviewPointsCard > matches snapshot with full data 1`] = `
<div>
  <div
    class="flex flex-col w-full h-full rounded-[10px] bg-purple_medium/[.05] p-3 gap-3"
  >
    <p
      class="text-sm font-medium font-normal"
    >
      My PX Overview
    </p>
    <div
      class="flex flex-col w-full gap-2.5"
    >
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="pillarx-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='6'%20viewBox='0%200%2016%206'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_1577_2383)'%3e%3cpath%20opacity='0.5'%20d='M15.5213%201.57674H16.0001V0.600006H15.5213C15.081%200.600006%2014.6584%200.770336%2014.3599%201.06575L13.6152%201.80829C13.5236%201.89877%2013.3935%201.952%2013.2605%201.952C13.1275%201.952%2012.9975%201.89877%2012.9059%201.80829L12.1612%201.06575C11.8627%200.770336%2011.4401%200.600006%2010.9997%200.600006H10.5269V1.57674H10.9997C11.1357%201.57674%2011.2628%201.62997%2011.3544%201.72046L12.0784%202.4417L11.3544%203.16294C11.2628%203.25343%2011.1327%203.30666%2010.9968%203.30666H10.5239V4.2834H10.9968C11.4371%204.2834%2011.8597%204.11307%2012.1582%203.81765L12.9029%203.07512C12.9946%202.98463%2013.1246%202.9314%2013.2576%202.9314C13.3906%202.9314%2013.5206%202.98463%2013.6122%203.07512L14.3599%203.81765C14.6584%204.11307%2015.081%204.2834%2015.5213%204.2834H16.0001V3.30666H15.5213C15.3854%203.30666%2015.2583%203.25343%2015.1667%203.16294L14.4427%202.4417L15.1667%201.72046C15.2583%201.62997%2015.3884%201.57674%2015.5213%201.57674Z'%20fill='%238A77FF'/%3e%3cpath%20d='M1.03333%205.81735H0.0610335C0.0285252%205.81735%200.00488281%205.7934%200.00488281%205.76679V4.32962C0.00488281%203.29167%200.941713%202.44801%202.09428%202.44801C2.43119%202.44801%202.76513%202.52253%203.06362%202.66092L6.43266%204.30301C6.82276%204.49463%207.28379%204.49197%207.66798%204.29769C8.08468%204.08744%208.32997%203.67758%208.32997%203.24643V2.5518C8.32997%201.91306%207.75368%201.39409%207.04441%201.39409H0.0639888C0.0314805%201.39409%200.00783811%201.37014%200.00783811%201.34352V0.47058C0.00783811%200.441305%200.0344358%200.420013%200.0639888%200.420013H7.04145C8.35065%200.420013%209.41161%201.37546%209.41161%202.55446V3.23578C9.41161%204.01292%208.97422%204.74747%208.22948%205.13603C7.48475%205.5246%206.64544%205.52194%205.91253%205.16531L2.55235%203.52588C2.41641%203.462%202.25682%203.42741%202.09428%203.42741C1.54164%203.42741%201.08948%203.83194%201.08948%204.33229V5.76945C1.08948%205.79872%201.06288%205.82001%201.03333%205.82001V5.81735Z'%20fill='%238A77FF'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_1577_2383'%3e%3crect%20width='16'%20height='6'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            PX Points
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          1,234
           
          <span
            class="font-semibold text-white/[.5]"
          >
            PX
          </span>
        </p>
      </div>
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="current-rank-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cg%20clip-path='url(%23clip0_1577_2405)'%3e%3cpath%20opacity='0.4'%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M2.3999%2014.4C2.3999%2013.9582%202.71331%2013.6%203.0999%2013.6H12.8999C13.2865%2013.6%2013.5999%2013.9582%2013.5999%2014.4C13.5999%2014.8418%2013.2865%2015.2%2012.8999%2015.2H3.0999C2.71331%2015.2%202.3999%2014.8418%202.3999%2014.4Z'%20fill='%238A77FF'/%3e%3cpath%20d='M6.96961%201.51463C7.14292%201.0818%207.54939%200.799988%207.99999%200.799988C8.45074%200.799988%208.85699%201.08193%209.03038%201.51455L10.5707%205.36044C10.6669%205.60075%2010.5977%205.87835%2010.4012%206.03961C10.1842%206.21776%2010.0465%206.49196%2010.0465%206.79999C10.0465%207.33449%2010.4629%207.76773%2010.9767%207.76773C11.4904%207.76773%2011.9069%207.33449%2011.9069%206.79999C11.9069%206.69688%2011.8916%206.59859%2011.8636%206.50683C11.7957%206.28393%2011.8629%206.04036%2012.0341%205.88888L13.9344%204.20787C14.4266%203.7724%2015.1643%203.79633%2015.6284%204.26555C15.9609%204.6017%2016.0849%205.10344%2015.9405%205.56561L15.939%205.57058L14.0787%2011.3473C13.7992%2012.2151%2013.0179%2012.8%2012.1382%2012.8H3.86179C2.98205%2012.8%202.20075%2012.2151%201.92129%2011.3473L0.0594687%205.5656C-0.0849019%205.10344%200.0391005%204.6017%200.371607%204.26555C0.835751%203.79633%201.57328%203.77241%202.06554%204.20786L3.96588%205.88885C4.13714%206.04033%204.20432%206.28392%204.13633%206.50682C4.10835%206.59856%204.09304%206.69685%204.09304%206.79999C4.09304%207.33449%204.50951%207.76773%205.02327%207.76773C5.53702%207.76773%205.95349%207.33449%205.95349%206.79999C5.95349%206.49198%205.81581%206.21777%205.59875%206.03959C5.4023%205.87833%205.33304%205.60075%205.42928%205.36043L6.96961%201.51463Z'%20fill='%238A77FF'/%3e%3c/g%3e%3cdefs%3e%3cclipPath%20id='clip0_1577_2405'%3e%3crect%20width='16'%20height='16'%20fill='white'/%3e%3c/clipPath%3e%3c/defs%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Current Rank
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          <span
            class="font-semibold text-white/[.5]"
          >
            #
          </span>
          5
        </p>
      </div>
    </div>
    <p
      class="text-sm font-medium font-normal"
    >
      Weekly
    </p>
    <div
      class="flex flex-col w-full gap-2.5"
    >
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="pillarx-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20fill-rule='evenodd'%20clip-rule='evenodd'%20d='M7.11805%201.96958C7.55465%201.12122%208.77852%201.12122%209.21512%201.96958L10.6861%204.8282C10.7105%204.87563%2010.7564%204.90864%2010.8095%204.91702L14.007%205.42134C14.9554%205.57093%2015.3333%206.72184%2014.6551%207.39531L12.3661%209.66831C12.3281%209.70598%2012.3107%209.75924%2012.3191%209.81178L12.8236%2012.9841C12.9731%2013.9243%2011.9832%2014.6359%2011.1269%2014.2039L8.24292%2012.7489C8.19498%2012.7247%208.13819%2012.7247%208.09025%2012.7489L5.20627%2014.2039C4.34995%2014.6359%203.36008%2013.9243%203.5096%2012.9841L4.01414%209.81178C4.0225%209.75924%204.00502%209.70598%203.96711%209.66831L1.67804%207.39531C0.999871%206.72184%201.37779%205.57093%202.32619%205.42134L5.5237%204.91702C5.57678%204.90864%205.62263%204.87563%205.64704%204.8282L7.11805%201.96958Z'%20fill='%238A77FF'/%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Weekly Rank
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          <span
            class="font-semibold text-white/[.5]"
          >
            #
          </span>
          10
        </p>
      </div>
      <div
        class="flex w-full justify-between items-center"
      >
        <div
          class="flex gap-2 items-center"
        >
          <div
            class="flex h-6 w-6 items-center justify-center rounded-[5px] bg-purple_medium/[.05]"
          >
            <img
              alt="current-rank-icon"
              src="data:image/svg+xml,%3csvg%20width='16'%20height='16'%20viewBox='0%200%2016%2016'%20fill='none'%20xmlns='http://www.w3.org/2000/svg'%3e%3cpath%20d='M7%201.16669C8.5474%201.16669%209.80427%202.40717%209.832%203.94794H4.16797C4.19576%202.40717%205.4526%201.16669%207%201.16669Z'%20fill='%238A77FF'/%3e%3cpath%20opacity='0.4'%20d='M3.63086%203.05467C3.75694%202.60425%203.97053%202.19071%204.25261%201.83331H3.44141C2.3694%201.83331%201.50011%202.65423%201.5%203.66665V10.3763C1.49998%2011.2849%201.49999%2012.0178%201.57747%2012.5944C1.65815%2013.1944%201.83121%2013.6996%202.23242%2014.1009C2.62076%2014.4508%203.17638%2014.6802%203.73893%2014.7558C4.31554%2014.8334%205.04833%2014.8333%205.95703%2014.8333H10.043C10.9517%2014.8333%2011.6845%2014.8334%2012.2611%2014.7558C12.8611%2014.6752%2013.4392%2014.426%2013.8405%2014.0247C14.1903%2013.6364%2014.3469%2013.1569%2014.4225%2012.5944C14.4651%2012.2778%2014.4829%2011.914%2014.4915%2011.5H12.6667C12.4236%2011.5%2012.133%2011.4848%2011.9453%2011.4486C11.46%2011.3184%2011.0739%2010.9552%2010.9134%2010.4837C10.8581%2010.2769%2010.8333%209.94431%2010.8333%209.66665C10.8333%209.38891%2010.8581%209.05631%2010.9134%208.84958C11.074%208.37811%2011.46%208.01478%2011.9453%207.88471C12.133%207.84845%2012.4236%207.83331%2012.6667%207.83331H14.4915C14.4829%207.41931%2014.4651%207.05551%2014.4225%206.73891C14.3469%206.17642%2014.1903%205.69688%2013.8405%205.30857C13.4935%204.96163%2012.9959%204.70937%2012.4993%204.61587C12.4983%204.28285%2012.4917%204.01122%2012.4427%203.77147L12.4115%203.64516C12.1871%202.85445%2011.5601%202.22588%2010.7467%201.96417L10.5814%201.91665C10.3413%201.85592%2010.0743%201.84037%209.74933%201.83592C10.0536%202.22214%2010.2774%202.67389%2010.3978%203.16665C10.8171%203.34961%2011.1235%203.73991%2011.1869%204.21027C11.1986%204.29749%2011.1986%204.40137%2011.1986%204.60619V4.61521H3.57422C3.14344%204.61504%202.79427%204.26544%202.79427%203.83461C2.79451%203.40399%203.14358%203.05485%203.57422%203.05467H3.63086Z'%20fill='%238A77FF'/%3e%3cpath%20d='M14.4999%2010.5H12.6666C12.4098%2010.5%2012.248%2010.4908%2012.248%2010.4908C12.1049%2010.4999%2011.8502%2010.2998%2011.8502%2010.1295C11.8501%2010.0803%2011.8333%2010.0085%2011.8333%209.66665C11.8333%209.32458%2011.8375%209.25131%2011.8502%209.20378C11.8503%209.06751%2012.0859%208.84251%2012.248%208.84245C12.3046%208.83585%2012.4098%208.83331%2012.6666%208.83331H14.4999V10.5Z'%20fill='%238A77FF'/%3e%3c/svg%3e"
            />
          </div>
          <p
            class="text-sm font-medium font-normal text-white/[.5]"
          >
            Earned This Week
          </p>
        </div>
        <p
          class="text-sm font-medium font-semibold text-white"
        >
          345
           
          <span
            class="font-semibold text-white/[.5]"
          >
            PX
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
